integrations/discord/websockets/socketmode.go

```diff
@@ -0,0 +1,122 @@
+package websockets
+
+import (
+	"context"
+	"sync"
+
+	"github.com/bwmarrin/discordgo"
+	"go.uber.org/zap"
+
+	"go.autokitteh.dev/autokitteh/integrations/internal/extrazap"
+	"go.autokitteh.dev/autokitteh/internal/kittehs"
+	"go.autokitteh.dev/autokitteh/sdk/sdkservices"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+type handler struct {
+	logger        *zap.Logger
+	vars          sdkservices.Vars
+	dispatcher    sdkservices.Dispatcher
+	integration   sdktypes.Integration
+	integrationID sdktypes.IntegrationID
+}
+
+func NewHandler(l *zap.Logger, v sdkservices.Vars, d sdkservices.Dispatcher, i sdktypes.Integration) handler {
+	return handler{
+		logger:        l,
+		vars:          v,
+		dispatcher:    d,
+		integration:   i,
+		integrationID: i.ID(),
+	}
+}
+
+var (
+	webSocketClients = make(map[string]*discordgo.Session)
+
+	mu = &sync.Mutex{}
+)
+
+func (h handler) OpenSocketModeConnection(botID, botToken string) {
```

Comment:
why do we need to cache connections?

Reply:
I'm guessing to keep their websockets alive. My question (maybe that's what Itay is really asking): is that really necessary? In other words, if we just create the objects without saving them, are the websockets garbage-collected?

Reply:
I checked. I don't think we do. I removed caching.